feat: add aws_security_group_inline_rules rule (#793)
terraform-linters/tflint-ruleset-aws: docs/rules/aws_security_group_egress_and_ingress_blocks_deprecated.md

# aws_security_group_egress_and_ingress_blocks_deprecated

> Avoid using the `ingress` and `egress` arguments of the `aws_security_group` resource to configure in-line rules, as they have difficulties managing multiple CIDR blocks and lack unique IDs, tags, and descriptions. To prevent these issues, follow the current best practice of using the `aws_vpc_security_group_egress_rule` and `aws_vpc_security_group_ingress_rule` resources, with one CIDR block per rule.

Reviewer: ```suggestion
Disallow `ingress` and `egress` arguments of the `aws_security_group` resource
```

You can explain why this rule is necessary in the "Why" section below.

## Example

```hcl
resource "aws_security_group" "foo" {
  name = "test"

  egress {
    from_port   = 0
    to_port     = 0
    protocol    = "-1"
    cidr_blocks = ["0.0.0.0/0"]
  }

  ingress {
    from_port   = 443
    to_port     = 443
    protocol    = "tcp"
    cidr_blocks = ["0.0.0.0/0"]
  }
}
```

```
$ tflint
2 issue(s) found:

Notice: Replace this egress block with aws_vpc_security_group_egress_rule. Otherwise, rule conflicts, perpetual differences, and result in rules being overwritten may occur. (aws_security_group_egress_and_ingress_blocks_deprecated)

  on test.tf line 4:
   4:   egress {
   
Notice: Replace this ingress block with aws_vpc_security_group_ingress_rule. Otherwise, rule conflicts, perpetual differences, and result in rules being overwritten may occur. (aws_security_group_egress_and_ingress_blocks_deprecated)

  on test.tf line 11:
   11:   ingress {
   
```

## Why

> Refrain from using the `ingress` and `egress` arguments of the `aws_security_group` resource for in-line rules, as they have difficulties managing multiple CIDR blocks and historically lack unique IDs, tags, and descriptions. To prevent these issues, follow the best practice of using the `aws_vpc_security_group_egress_rule` and `aws_vpc_security_group_ingress_rule` resources, with one CIDR block per rule.

Reviewer: Please include a link to what "best practices" point to, perhaps this one?
https://registry.terraform.io/providers/hashicorp/aws/5.82.2/docs/resources/security_group

> Avoid using the `aws_security_group` resource with in-line rules (using the ingress and egress arguments) alongside the `aws_vpc_security_group_egress_rule`, `aws_vpc_security_group_ingress_rule`, or `aws_security_group_rule` resources. This practice can lead to rule conflicts, perpetual differences, and rules being overwritten.

Reviewer: These sentences appear to be copied from official references, but they feel a bit odd to include in this section.
Can you tidy up these sentences and rewrite them to explain why this rule is intended to disallow inline rules?

Author: Seems to be a duplication. Paragraph reworked.
